ansible/roles/vm_set/tasks/start_k8s.yml

```diff
@@ -0,0 +1,40 @@
+- name: Install cloud-image-utils
+  apt:
+    name: cloud-image-utils
+    state: present
+  become: yes
+
+- name: Ensure {{ k8s_root_path }} exists
+  file: path={{ k8s_root_path }} state=directory
+
+- name: Find current server group
+  set_fact: current_server={{ group_names | extract_by_prefix('k8s_server_') }}
+
+- name: Extract VM names from the inventory
+  set_fact: kvm_hosts={{ groups[current_server] | filter_by_prefix('kvm') | sort}}
+
+- name: Create directory for Ubuntu VM image
+  file: path={{ k8s_root_path }}/images state=directory mode=0755
+
+- name: Check local cloud image 
+  stat: 
+    path: "{{ k8s_root_path }}/images/{{ k8s_hdd_image_filename }}"
+    get_checksum: no 
+  register: cloudimg_local_stat 
+
+- name: Download Ubuntu 18.04 hdd image
+  get_url: url="{{ k8s_vm_images_url }}/{{ k8s_hdd_image_filename }}?{{ k8s_vmimage_saskey }}" dest="{{ k8s_root_path }}/images/{{ k8s_hdd_image_filename }}"
+  when: not cloudimg_local_stat.stat.exists and not k8s_skip_image_downloading
+  become: yes
```

Comment:
You could merge `stat` and `get_url` into just `get_url`.

ref: [https://docs.ansible.com/ansible/latest/collections/ansible/builtin/get_url_module.html](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/get_url_module.html)
> the file will only be downloaded if the destination does not exist

Reply:
Even when `force` in `get_url` module is set to no (by default and specified), this job takes a long time using just `get_url` module even when the destination already exists. I double checked that the file is there, and observed that it still tries to connect to `https://acsbe.blob.core.windows.net/vmimages/bionic-server-cloudimg-amd64.qcow2?use_own_value` when the file is already present. If we leave it as is with `stat` and `get_url` module, it would be similar to the check for veos VM